WalletWasabi.Fluent/ViewModels/AddWallet/AddWalletPageViewModel.cs

```diff
@@ -22,68 +23,71 @@ namespace WalletWasabi.Fluent.ViewModels.AddWallet;
+		ConnectHardwareWalletCommand = ReactiveCommand.Create(OnConnectHardwareWallet);
 
-		NextCommand = ReactiveCommand.Create(OnNext, canExecute);
+		ImportWalletCommand = ReactiveCommand.CreateFromTask(async () => await OnImportWalletAsync());
 
-		OpenCommand = ReactiveCommand.Create(
-			async () =>
-		{
-			MainViewModel.Instance.IsOobeBackgroundVisible = true;
+		RecoverWalletCommand = ReactiveCommand.Create(OnRecoverWallet);
 
-			await NavigateDialogAsync(this, NavigationTarget.DialogScreen);
+		OpenCommand = ReactiveCommand.Create(async () => await NavigateDialogAsync(this, NavigationTarget.DialogScreen));
+	}
 
-			MainViewModel.Instance.IsOobeBackgroundVisible = false;
-		});
+	public ICommand CreateWalletCommand { get; }
 
-		this.ValidateProperty(x => x.WalletName, errors => ValidateWalletName(errors, WalletName));
-	}
+	public ICommand ConnectHardwareWalletCommand { get; }
 
-	private void OnNext()
+	public ICommand ImportWalletCommand { get; }
+
+	public ICommand RecoverWalletCommand { get; }
+
+	private void OnCreateWallet()
 	{
-		Navigate().To(new SelectWalletCreationOptionViewModel(WalletName));
+		Navigate().To(new WalletNamePageViewModel(WalletCreationOption.AddNewWallet));
 	}
 
-	private static void ValidateWalletName(IValidationErrors errors, string walletName)
+	private void OnConnectHardwareWallet()
 	{
-		string walletFilePath = Path.Combine(Services.WalletManager.WalletDirectories.WalletsDir, $"{walletName}.json");
-
-		if (string.IsNullOrEmpty(walletName))
-		{
-			return;
-		}
-
-		if (walletName.IsTrimmable())
-		{
-			errors.Add(ErrorSeverity.Error, "Leading and trailing white spaces are not allowed!");
-			return;
-		}
+		Navigate().To(new WalletNamePageViewModel(WalletCreationOption.ConnectToHardwareWallet));
+	}
 
-		if (File.Exists(walletFilePath))
+	private async Task OnImportWalletAsync()
+	{
+		try
 		{
-			errors.Add(
-				ErrorSeverity.Error,
-				$"A wallet named {walletName} already exists. Please try a different name.");
-			return;
+			var filePath = await FileDialogHelper.ShowOpenFileDialogAsync("Import wallet file", new[] { "json" });
+
+			if (filePath is null)
+			{
+				return;
+			}
+
+			var walletName = Path.GetFileNameWithoutExtension(filePath);
+
+			var validationError = WalletHelpers.ValidateWalletName(walletName);
+			if (validationError is { } err)
+			{
+				Navigate().To(new WalletNamePageViewModel(WalletCreationOption.ImportWallet, filePath));
+				return;
+			}
+		
+			var keyManager = await ImportWalletHelper.ImportWalletAsync(Services.WalletManager, walletName, filePath);
+			Navigate().To(new AddedWalletPageViewModel(keyManager));
 		}
-
-		if (!WalletGenerator.ValidateWalletName(walletName))
+		catch (Exception ex)
 		{
-			errors.Add(ErrorSeverity.Error, "Selected Wallet is not valid. Please try a different name.");
+			Logger.LogError(ex);
+			await ShowErrorAsync(Title, ex.ToUserFriendlyString(),
+				"The wallet file was not valid or compatible with Wasabi.");
```

Comment:
`Title` -> `Import wallet`
`Message` -> `Wasabi was unable to import your wallet`

For the consistency.